OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs

```diff
@@ -197,6 +197,60 @@ internal static class DreamProcNativeRoot {
             var invisibility = bundle.GetArgument(24, "invisibility");
             var suffix = bundle.GetArgument(25, "suffix");
 
+            if((flags & AnimationFlags.AnimationRelative) != 0){
+                // This works for maptext_x/y/width/height, pixel_x/y/w/z, luminosity, layer, alpha, transform, and color. For transform and color, the current value is multiplied by the new one. Vars not in this list are simply changed as if this flag is not present.
+                if(!pixelX.IsNull)
+                    pixelX = new(pixelX.UnsafeGetValueAsFloat() + obj.GetVariable("pixel_x").UnsafeGetValueAsFloat());
```

Comment:
You should get `obj`'s appearance using AtomManager and access the values directly instead of doing `obj.GetVariable(...)` which is much less efficient.